assets/src/blocks/setupCustomSidebar.js

```diff
@@ -0,0 +1,49 @@
+import {registerPlugin} from '@wordpress/plugins';
+import {OpenGraphSidebar} from '../components/Sidebar/OpenGraphSidebar';
+
+const sidebarsForPostType = postType => {
+  switch (postType) {
+  case 'campaign':
+    return [
+      OpenGraphSidebar,
+    ];
+  case 'p4_action':
+    return [
+      OpenGraphSidebar,
+    ];
+  case 'page':
+    return [
+      OpenGraphSidebar,
+    ];
+  case 'post':
+    return [
+      OpenGraphSidebar,
+    ];
+  default:
+    return null;
+  }
+};
+
+export const setupCustomSidebar = () => {
+  let currentPostType = null;
+  // Only subscribing after DOMContentLoaded avoids the troubles originating from wp.data emitting null values before that point.
+  document.addEventListener('DOMContentLoaded', () => {
+    wp.data.subscribe(() => {
+      const newPostType = wp.data.select('core/editor').getCurrentPostType();
+      if (newPostType === currentPostType) {
+        return;
+      }
+
+      currentPostType = newPostType;
```

Comment:
Why do we assign `currentPostType` here?

Reply:
Initially the `const newPostType = wp.data.select('core/editor').getCurrentPostType();` returns `null` value, in subsequent call it returns the actual post type value.